juddges/data/qa_pairs_json_parser.py

```diff
@@ -0,0 +1,62 @@
+import re
+from json.decoder import JSONDecodeError
+from typing import Any, Callable, Dict, List
+
+from langchain_core.exceptions import OutputParserException
+from langchain_core.output_parsers import JsonOutputParser
+from langchain_core.output_parsers.json import parse_partial_json  # type: ignore
+from langchain_core.outputs import Generation
+from langchain_core.utils.json import _parse_json
+
+CUSTOM_PARSE_JSON_MARKDOWN = re.compile(
```

Comment:
check https://python.langchain.com/v0.1/docs/modules/model_io/output_parsers/types/json/

Reply:
original `JsonOutputParser` parses output successfully always when prompt does not contain any other JSON structure, and the `parse_json_markdown` pattern is greedy which is not nice

https://github.com/langchain-ai/langchain/pull/20305/files#diff-7736e09b3c57a6d2f3803974d6022d46a6c7cb44f8688af5a72bea77d9db5124L139